checked_message_test.go

```diff
@@ -28,31 +28,35 @@ import (
 )
 
 func TestJSONLoggerCheck(t *testing.T) {
-	withJSONLogger(t, opts(InfoLevel), func(jl *jsonLogger, output func() []string) {
-		assert.False(t, jl.Check(DebugLevel, "Debug.").OK(), "Expected CheckedMessage to be not OK at disabled levels.")
+	withJSONLogger(t, opts(InfoLevel), func(logger Logger, buf *testBuffer) {
+		assert.False(
+			t,
+			logger.Check(DebugLevel, "Debug.").OK(),
+			"Expected CheckedMessage to be not OK at disabled levels.",
+		)
 
-		cm := jl.Check(InfoLevel, "Info.")
+		cm := logger.Check(InfoLevel, "Info.")
 		require.True(t, cm.OK(), "Expected CheckedMessage to be OK at enabled levels.")
 		cm.Write(Int("magic", 42))
 		assert.Equal(
 			t,
-			`{"msg":"Info.","level":"info","ts":0,"fields":{"magic":42}}`,
-			output()[0],
+			`{"magic":42,"msg":"Info.","level":"info"}`,
```

Comment:
While the JSON output isn't intended for users, it will often be consumed by humans. The current ordering of keys isn't very human-friendly, I'd prefer: time, level, msg, fields. (specifically, the time, level and message should be before the fields).

Since we're copying the bytes in clone anyway, I don't think it'll be any less efficient to create a new buffer, append the time/level/message fields before copying over the fields buffer.